docs/technical/installation_controller.md

```diff
@@ -0,0 +1,38 @@
+# Installation Controller
+
+The reconciliation flow of the installation controller is described in detail [here](./reconciliation_flow.md).
+It executes one of the following main operations:
+
+- a deletion,
+- a [standard reconcile](#standard-reconcile),
+- a [force reconcile](#force-reconcile),
+- or a [re-check of completed installation](#re-check-completed-installation).
+
+The diagrams below illustrate these flows. First, the decision which main operation will be executed: 
+
+![inst-main-reconcile](images/inst-main-reconcile.png)
+
+
+## Standard Reconcile
+
+![inst-standard-reconcile](images/inst-standard-reconcile.png)
```

Comment:
first condition: sub-objects busy: does it mean that their phase is progressing?
condition update allowed returns an error, does this map to a function call? Otherwise it would be consistent to make it a yes/no condition

Reply:
Sub-objects busy: it can be a progressing phase, but also if the spec was updated and the controller has not yet reconciled (observed generation != generation). I.e. working, or there is work to do which was not yet started. 

Update allowed: actually at first I used yes/no. Then Achim pointed out that the [method UpdateAllowed](https://github.com/gardener/landscaper/blob/e08eafcb1f1320d8935f01d4d09dda7853263513/pkg/landscaper/installations/reconcilehelper/reconcilehelper.go#L89) does not return a boolean, but an error instead of no.